evalml/data_checks/mismatched_series_length_data_check.py

```diff
@@ -0,0 +1,154 @@
+            >>> mismatched_series_length_check = MismatchedSeriesLengthDataCheck("series_id")
+            >>> assert mismatched_series_length_check.validate(X) == [
+            ...      {
+            ...         "message": "Series ID ['0'] do not match the majority length of the other series, which is 20",
+            ...         "data_check_name": "MismatchedSeriesLengthDataCheck",
+            ...         "level": "warning",
+            ...         "details": {
+            ...             "columns": None,
+            ...             "rows": None,
+            ...             "series_id": ['0'],
+            ...             "majority_length": 20
+            ...         },
+            ...         "code": "MISMATCHED_SERIES_LENGTH",
+            ...         "action_options": [],
+            ...     }
+            ... ]
+
+            If MismatchedSeriesLengthDataCheck is passed in an invalid series_id column name, then an error will be raised.
+
+            >>> X = pd.DataFrame(
+            ...     {
+            ...         "date": pd.date_range(start="1/1/2018", periods=20).repeat(5),
+            ...         "series_id": pd.Series(list(range(5)) * 20, dtype="str"),
+            ...         "feature_a": range(100),
+            ...         "feature_b": reversed(range(100)),
+            ...     },
+            ... )
+            >>> X = X.drop(labels=0, axis=0)
+            >>> mismatched_series_length_check = MismatchedSeriesLengthDataCheck("not_series_id")
+            >>> assert mismatched_series_length_check.validate(X) == [
+            ...      {
+            ...         "message": "series_id 'not_series_id' does not match the series_id column of the dataset.",
+            ...         "data_check_name": "MismatchedSeriesLengthDataCheck",
+            ...         "level": "error",
+            ...         "details": {
+            ...             "columns": None,
+            ...             "rows": None,
+            ...             "series_id": "not_series_id",
+            ...         },
+            ...         "code": "MULTISERIES_TIMESERIES_SERIES_ID_NOT_IN_COL",
+            ...         "action_options": [],
+            ...     }
+            ... ]
+        """
+        messages = []
+        if self.series_id not in X:
+            messages.append(
+                DataCheckError(
+                    message=f"""series_id '{self.series_id}' does not match the series_id column of the dataset.""",
+                    data_check_name=self.name,
+                    message_code=DataCheckMessageCode.MULTISERIES_TIMESERIES_SERIES_ID_NOT_IN_COL,
+                    details={"series_id": self.series_id},
+                ).to_dict(),
+            )
+            return messages
+
+        # gets all the number of entries per series_id
+        series_id_len = {}
+        for id in X[self.series_id].unique():
+            series_id_len[id] = len(X[X[self.series_id] == id])
+
+        # get the majority length
```

Comment:
Is this faster than what I currently have or is it more concise?

Reply:
Its more concise but I _believe_ its the same time complexity. You can double check here:
- https://github.com/python/cpython/blob/3.11/Lib/statistics.py
- https://github.com/python/cpython/blob/3.11/Lib/collections/__init__.py#L624

Reply:
I think I'll stick to what I have currently since what I have lets me easily exit out early if all the series have the same length, which is something that Becca asked to do in one of her comments.